docs/source/future/core/nrunner.rst

```diff
@@ -53,54 +53,166 @@ Runnable
+  >>> runnable
+  <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
+
+The second way is through a JSON based file, which, for the lack of a
+better term, we're calling a (Runnable) "recipe".  The recipe file
+itself will look like::
+
+  {"kind": "exec", "uri": "/bin/true"}
+
+And example the code to create it::
+
+  >>> from avocado.core import nrunner
+  >>> runnable = nrunner.Runnable.from_recipe("/path/to/recipe.json")
+  >>> runnable
+  >>> <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
+
+The third way to create a Runnable, is even more internal.  Its usage
+is **discouraged**, unless you are creating a tool that needs to
+create Runnables based on the user's input from the command line::
+
+  >>> from avocado.core import nrunner
+  >>> runnable = nrunner.Runnable.from_args({kind: 'exec', uri: '/bin/true'})
+  >>> runnable
+  >>> <Runnable kind="exec" uri="/bin/true" args="()" kwargs="{}" tags="None">
 
 Runner
 ~~~~~~
 
-A runner is an active entity that acts on the information of a
-runnable.  A runner will usually be tied to an specific kind of
-runnable, and will to able to act upon the specific information that
-runnable kind provides.
+A Runner, within the context of the N(ext)Runner architecture, is an
+active entity that acts on the information that a runnable contains,
+and quite simply, should be able to run what the Runnable describes.
+
+A Runner will usually be tied to a specific ``kind`` of Runnable.
+That type of relationship (Runner is capable of running kind "foo"
+and Runnable is of the same kind "foo") is the expected mechanism that
+will be employed when selecting a Runner.
+
+A Runner can take different forms, depending on which layer one is
+interacting with.  At the lowest layer, a Runner may be a Python class
+that inherits from :class:`avocado.core.nrunner.BaseRunner`, and
+implements at least a matching constructor method, and a ``run()``
+method that should yield dictionary(ies) as result(s).
+
+At a different level, a runner can take the form of an executable that
+follows the ``avocado-runner-$KIND`` naming pattern and conforms to a
+given interface/behavior, including accepting standardized command
+line arguments and producing standardized output.
+
+.. tip:: for a very basic example of the interface expected, refer to
+         ``selftests/functional/test_nrunner_interface.py`` on the
+         Avocado source code tree.
 
-The following pseudo-code may help to illustrate that::
+Runner output
++++++++++++++
 
-  runnable_instance = create_runnable('exec', uri='/bin/sleep')
-  if runnable_instance.kind == 'exec':
-     runner = create_runner_exec(runnable_instance)
+A Runner should, if possible, produce status information on the
+progress of the execution of a Runnable.  While the Runner is
+executing what a Runnable describes, should it produce interesting
+information, the Runner should attempt to forward that along its
+generated status.
```

Comment:
Small suggestion: While the Runner is executing what a Runnable describes, should it produce **relevant** information, the Runner should attempt to forward that along **with** its generated status.

Reply:
Makes sense, I'll queue that change for the next round. Thanks!